pkg/gofr/context_test.go

```diff
@@ -36,3 +38,56 @@ func Test_newContextSuccess(t *testing.T) {
 	assert.Equal(t, map[string]string{"key": "value"}, body, "TEST Failed \n unable to read body")
 	require.NoError(t, err, "TEST Failed \n unable to read body")
 }
+
+func TestContext_AddTrace(t *testing.T) {
+	ctxBase := context.Background()
+	ctx := Context{
+		Context: ctxBase,
+	}
+
+	span := ctx.Trace("Some Work")
+
+	defer span.End()
+
+	assert.NotEqual(t, ctxBase, ctx.Context)
+}
+
+func TestContext_WriteMessageToSocket(t *testing.T) {
+	t.Setenv("HTTP_PORT", "8005")
+
+	app := New()
+
+	server := httptest.NewServer(app.httpServer.router)
+	defer server.Close()
+
+	app.WebSocket("/ws", func(ctx *Context) (interface{}, error) {
+		err := ctx.WriteMessageToSocket("Hello! GoFr")
+		if err != nil {
+			return nil, err
+		}
+
+		// returning error here to close the connection to the websocket
+		// as the websocket close error is not caught because we are using no bind function here.
+		// this must not be necessary. We should put an actual check in handleWebSocketConnection method instead.
+		return nil, &websocket.CloseError{Code: websocket.CloseNormalClosure, Text: "Error closing"}
+	})
+
+	go app.Run()
+
+	wsURL := "ws" + server.URL[len("http"):] + "/ws"
+
+	// Create a WebSocket client
+	ws, resp, err := websocket.DefaultDialer.Dial(wsURL, nil)
+	require.NoError(t, err)
+
+	defer resp.Body.Close()
+	defer ws.Close()
+
+	_, message, err := ws.ReadMessage()
+
+	require.NoError(t, err)
+
+	// Read the response
+	expectedResponse := "Hello! GoFr"
+	assert.Equal(t, expectedResponse, string(message))
```

Comment:
we are not actually `reading` the response here, we are validating the expected and actual msg.
pls update/remove the comment.